dip-0031.md

```diff
@@ -0,0 +1,120 @@
+<pre>
+  DIP: 0031
+  Title: Platform Proof of Service
+  Chain: Core
+  Author(s): Pasta
+  Special-Thanks:
+  Comments-Summary: No comments yet.
+  Status: Proposed
+  Type: Standard
+  Created: 2024-09-12
+  License: MIT License
+</pre>
+
+# Table of Contents
+
+* [Abstract](#abstract)
+* [Motivation](#motivation)
+* [Prior Work](#prior-work)
+* [PlatformBan P2P Message](#platformban-p2p-message)
+* [Validation](#validation)
+* [Processing](#processing)
+* [Effect](#effect)
+* [Signing Process](#signing-process)
+* [Ban RPC command](#ban-rpc-command)
+* [Copyright](#copyright)
+
+## Abstract
+
+Core will introduce a new P2P message `platformban`, which, when properly signed and received by a
+masternode, will result in a masternode voting against the target in all future DKG sessions until
+that targeted masternode has been successfully banned.
+
+## Motivation
+
+To ensure that evonodes provide service on Dash Platform and that non-functioning evonodes do not
+participate in the Platform validator set, Platform needs a way to exclude faulty evonodes from
+future Platform validator sets / quorums. Since these nodes are not properly providing service, they
+should be proof-of-service (PoSe) banned. This DIP outlines a method for Platform to securely notify
+Core that a node should be banned.
+
+## Prior Work
+
+* [DIP-0006: Long-Living Masternode
+  Quorums](https://github.com/dashpay/dips/blob/master/dip-0006.md)  
+* [DIP-0007: LLMQ Signing Requests /
+  Sessions](https://github.com/dashpay/dips/blob/master/dip-0007.md)
+
+## PlatformBan P2P Message
+
+Core will introduce a new P2P message, which will facilitate the PoSe banning of nodes identified as
+not providing proper service on Platform. This P2P message will be broadcast via the inventory
+system and introduced in Protocol Version `70235`.
+
+| Field | Type | Size(bytes) | Description |
+| ----- | ----- | ----- | ----- |
+| protxHash | uint256 | 32 | The protxHash of the masternode being banned |
+| signedHeight | int32_t | 4 | The height at which this message was signed by the platform quorum |
+| quorumHash | uint256 | 32 | The quorumHash of the quorum which signed the message |
+| blsSignature | byte[] | 96 | The BLS signature over the message, by the quorum identified by quorumHash |
+
+Note: this P2P message intentionally excludes an internal version; instead, it is versioned via the
+Protocol Version.
+
+## Validation
+
+When received, a node must validate the message. It should do this by:
+
+1. Check if the node is synced  
+   * Penalty: skip validation of message, do not forward, do not mark as received in inventory  
+2. Validate the protxHash is in the masternodeList  
+   * Penalty: small P2P penalty (1), as the masternode may have very recently been removed  
```

Comment:
We should use the term evonode everywhere instead of masternode and do not mix both terms.

Reply:
I made some updates to improve terminology. The doc does relate to both regular masternodes and evonodes in some parts, so I couldn't just convert them all to evonode.